core/cleaner/src/test/java/com/cognifide/aet/cleaner/processors/RemoveArtifactsProcessorTest.java

```diff
@@ -0,0 +1,160 @@
+
+    verify(artifactDAO, times(1)).removeArtifacts(any(DBKey.class), any(Set.class));
+    verify(artifactDAO, times(1)).getArtifactsId(any(DBKey.class));
+  }
+
+  @Test
+  public void check_ifRemoveArtifactsWasCalled_expectFalse() throws Exception {
+    when(cleanerContext.isDryRun()).thenReturn(true);
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(ONE_TO_SEVEN_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(ONE_TO_FIVE_ARTIFACTS_ID_SET);
+
+    removeArtifactsProcessor = new RemoveArtifactsProcessor(artifactDAO);
+    removeArtifactsProcessor.process(exchange);
+
+    verify(artifactDAO, times(0)).removeArtifacts(any(DBKey.class), any(Set.class));
+    verify(artifactDAO, times(1)).getArtifactsId(any(DBKey.class));
+  }
+
+  @Test
+  public void check_substractArtifactsSets_expectSetOfTwoVariables() throws Exception {
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(ONE_TO_SEVEN_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(ONE_TO_FIVE_ARTIFACTS_ID_SET);
+    ReferencedArtifactsMessageBody messageBody = (ReferencedArtifactsMessageBody) exchange.getIn()
+        .getBody();
+    assertEquals(SIX_TO_SEVEN_ARTIFACTS_ID_SET,
+        removeArtifactsProcessor.getArtifactsIdsToRemove(artifactDAO, messageBody));
+  }
+
+  @Test
+  public void check_substractArtifactsSets_expectEmptySet() throws Exception {
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(ONE_TO_FIVE_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(ONE_TO_SEVEN_ARTIFACTS_ID_SET);
+    ReferencedArtifactsMessageBody messageBody = (ReferencedArtifactsMessageBody) exchange.getIn()
+        .getBody();
+    assertEquals(EMPTY_ARTIFACTS_ID_SET,
+        removeArtifactsProcessor.getArtifactsIdsToRemove(artifactDAO, messageBody));
+  }
+
+  @Test
+  public void check_substractArtifactsSets_expectSetOfFiveVariables() throws Exception {
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(ONE_TO_SEVEN_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(SIX_TO_SEVEN_ARTIFACTS_ID_SET);
+
+    ReferencedArtifactsMessageBody messageBody = (ReferencedArtifactsMessageBody) exchange.getIn()
+        .getBody();
+
+    assertEquals(ONE_TO_FIVE_ARTIFACTS_ID_SET,
+        removeArtifactsProcessor.getArtifactsIdsToRemove(artifactDAO, messageBody));
+  }
+
+  @Test
+  public void check_substractArtifactsSetsWhenDbIsEmpty_expectEmptySet() throws Exception {
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(EMPTY_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(SIX_TO_SEVEN_ARTIFACTS_ID_SET);
+    ReferencedArtifactsMessageBody messageBody = (ReferencedArtifactsMessageBody) exchange.getIn()
+        .getBody();
+    assertEquals(EMPTY_ARTIFACTS_ID_SET,
+        removeArtifactsProcessor.getArtifactsIdsToRemove(artifactDAO, messageBody));
+  }
+
+  private void setArtifactsIdToKeep(Set<String> artifactsIdToRemove) {
```

Comment:
So this is to keep or remove :) ?

Reply:
It should be https://github.com/Cognifide/aet/pull/336/commits/af28aa4f87c820d637570ef77974675b12e0ff52 🎃